pennylane/gradients/finite_difference.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for computing the finite-difference gradient
+of a quantum tape.
+"""
+# pylint: disable=protected-access,too-many-arguments
+import numpy as np
+from scipy.special import factorial
+
+import pennylane as qml
+
+
+def finite_diff_stencil(n, order, form):
+    r"""Generate the finite difference stencil (shift and coefficients)
+    for various derivatives, accuracy, and form.

```

Comment:
I would not really know what this is TBH. A stencil means it is a pattern or "template"? What is accuracy and form?

Reply:
Ah, I stole the terminology from here: https://en.wikipedia.org/wiki/Finite_difference_coefficient#Arbitrary_stencil_points

This was originally a private function, so I did not put enough effort into the docstring 😿

Reply:
@mariaschuld to answer a question you had down below: I originally called this function `get_stencil()`, and it was just a private function inside this module.

However, when writing the unit tests, I needed to mock/spy on it, which required importing it into `qml.gradients`... which meant renaming it to `finite_diff_stencil` :(

Reply:
I have renamed it `finite_diff_coeffs`! I think this is clearer